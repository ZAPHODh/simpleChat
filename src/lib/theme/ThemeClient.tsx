'use client'

import { ThemeProvider } from 'styled-components'
import { theme } from './theme'

type ThemeClientProps = {
    children: React.ReactNode
}

export const ThemeClient = ({ children }: ThemeClientProps) => {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}
