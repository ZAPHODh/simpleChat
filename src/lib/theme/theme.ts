import { DefaultTheme } from 'styled-components'

export const theme: DefaultTheme = {
    colors: {
        white: '#ffffff ',
        darker: '#0B0B0D',
        dark: '#181619',
        normalDark: '#181B22',
        brightDark: '#272A31',
        gold: '#A76B09',
        brightGold: '#DDA74F',
    },
    imgBorderEffects: {
        borderThickness: '10px',
        offset: '20px',
        gap: '5px',
    },
}
