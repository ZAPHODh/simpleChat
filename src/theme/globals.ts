'use client'

import { Montserrat } from 'next/font/google'
import { createGlobalStyle } from 'styled-components'
import { css } from 'styled-components'

const montserrat = Montserrat({
    subsets: ['latin'],
    fallback: ['Georgia', 'ui-serif', 'serif'],
    weight: ['300', '700'],
})

export const GlobalStyles = createGlobalStyle`${() => css`
    html,
    body {
        width: 100%;
        max-width: 100vw;
        padding: 0;
        margin: 0;
        font-family: ${montserrat.style.fontFamily};
        overflow-x: hidden;
    }

    main {
        width: 100%;
        max-width: 100vw;
        padding: 0;
        margin: 0;
    }

    input {
        font-family: inherit;
    }

    a {
        color: inherit;
        text-decoration: none;
    }

    * {
        box-sizing: border-box;
    }

    button {
        font-family: inherit;
    }
`}`
