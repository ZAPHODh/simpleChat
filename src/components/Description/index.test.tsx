import { render } from '@testing-library/react'
import { Description } from '.'
import { ThemeClient } from '../../theme/ThemeClient'

describe('Description', () => {
    it('should render the paragraph', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Description>test</Description>
            </ThemeClient>
        )

        const heading = getByRole('paragraph')

        expect(heading).toBeInTheDocument()
    })
    it('should render the paragraph with max chars', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Description maxChars={2}>test</Description>
            </ThemeClient>
        )

        const heading = getByRole('paragraph')

        expect(heading).toHaveTextContent('te...')
    })
    it('should take a snapshot', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Description>test</Description>
            </ThemeClient>
        )

        const heading = getByRole('paragraph')

        expect(heading).toMatchSnapshot()
    })
})
