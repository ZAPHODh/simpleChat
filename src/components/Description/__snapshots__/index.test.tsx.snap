// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should take a snapshot 1`] = `
<p
  class="sc-69380e9b-0 cOccBQ"
>
  test
</p>
`;
